frontend/app/routes/$lang/_protected/applications/$id.tsx

```diff
@@ -75,40 +81,53 @@ export async function loader({ context: { session }, params, request }: LoaderFu
-    .join(', ');
-  const provincialTerritorialSocialProgram = allProvincialTerritorialSocialPrograms
-    .filter((provinceTerritorialProgram) => provinceTerritorialProgram.id === applicationDetails.dentalBenefits?.provincialTerritorialSocialProgram)
-    .map((obj) => getNameByLanguage(locale, obj))
-    .join(', ');
-
   const t = await getFixedT(request, handle.i18nNamespaces);
-  const meta = { title: t('gcweb:meta.title.template', { title: t('applications:view-application.page-title') }) };
+  const meta = { title: t('gcweb:meta.title.template', { title: t('applications:view-application.page-title', { year: viewApplication?.submittedOn ? extractDateParts(viewApplication.submittedOn).year : '' }) }) };
 
   const idToken: IdToken = session.get('idToken');
   auditService.audit('page-view.application-details', { userId: idToken.sub });
   instrumentationService.countHttpStatus('application-details.view', 200);
 
-  return json({ locale, applicationDetails, maritalStatus, mailingProvince, homeProvince, mailingCountry, homeCountry, communicationPreferenceLang, preferredLanguage, provincialTerritorialSocialProgram, federalSocialProgram, meta });
+  return json({
+    locale,
+    applicationDetails,
+    allFederalSocialPrograms,
+    allProvincialTerritorialSocialPrograms,
+    maritalStatus,
+    mailingProvince,
+    homeProvince,
+    mailingCountry,
+    homeCountry,
+    communicationPreferenceLang,
+    preferredLanguage,
+    meta,
+  });
 }
 
 export default function ViewApplication() {
   const { t, i18n } = useTranslation(handle.i18nNamespaces);
-  const { applicationDetails, locale, maritalStatus, mailingProvince, homeProvince, mailingCountry, homeCountry, preferredLanguage, provincialTerritorialSocialProgram, federalSocialProgram, communicationPreferenceLang } = useLoaderData<typeof loader>();
+  const { applicationDetails, locale, allFederalSocialPrograms, allProvincialTerritorialSocialPrograms, maritalStatus, mailingProvince, homeProvince, mailingCountry, homeCountry, preferredLanguage, communicationPreferenceLang } =
+    useLoaderData<typeof loader>();
 
   const dateOfBirth = applicationDetails.dateOfBirth ? toLocaleDateString(parseDateTimeString(applicationDetails.dateOfBirth), locale) : '';
   const partnerDateOfBirth = applicationDetails.partnerInformation?.dateOfBirth ? toLocaleDateString(parseDateTimeString(applicationDetails.partnerInformation.dateOfBirth), locale) : '';
+  const federalSocialProgram = allFederalSocialPrograms
+    .filter((federalProgram) => federalProgram.id === applicationDetails.dentalBenefits?.federalSocialProgram)
+    .map((obj) => getNameByLanguage(locale, obj))
+    .join(', ');
+  const provincialTerritorialSocialProgram = allProvincialTerritorialSocialPrograms
+    .filter((provinceTerritorialProgram) => provinceTerritorialProgram.id === applicationDetails.dentalBenefits?.provincialTerritorialSocialProgram)
+    .map((obj) => getNameByLanguage(locale, obj))
+    .join(', ');
```

Comment:
Finding the `federalSocialProgram` and `provincialTerritorialSocialProgram` for a particular application should be done in the `loader(..)` in my opinion. 

We don't need to be passing `allFederalSocialPrograms` and `allProvincialTerritorialSocialPrograms` to the component/client when the component is only display a single `federalSocialProgram` and `provincialTerritorialSocialProgram`.